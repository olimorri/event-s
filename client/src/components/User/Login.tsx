import React, { useState } from 'react';
import { Link as RouterLink } from 'react-router-dom';
import UsersApiService from '../../services/UsersApiService';
import auth from '../../utils/auth';
import {
  Flex,
  Box,
  FormControl,
  FormLabel,
  Input,
  Stack,
  Button,
  Heading,
  Text,
} from '@chakra-ui/react';
import { motion } from 'framer-motion'

const initialState = {
  email: '',
  password: '',
};


export default function Login(props) {
  const [state, setState] = useState(initialState);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setState((prevState) => ({
      ...prevState,
      [name]: value,
    }));
  };

  const handleSubmit = async (e) => {
    e.preventDefault();

    const { email, password } = state;
    const user = { email, password };
    const res = await UsersApiService.login(user);
    if (res.error) {
      alert(`${res.message}`);
      setState(initialState);
    } else {

      props.setIsAuthenticated(true);
      auth.login(() => props.history.push('/profile'));

    }
  };

  const validateForm = () => {
    return !state.email || !state.password;
  };

  return (
    <Flex minH={'100vh'} align={'center'} justify={'center'} bg={'custom.100'}
    bgImage="url('https://res.cloudinary.com/dujun1hoe/image/upload/v1615228154/event-s/gradient-background-26046-26731-hd-wallpapers.jpg_cenrqe.png')"  
    bgSize="100%"
    backgroundRepeat="no-repeat"
    >
      <motion.div
      initial={{
        opacity: 0,
        x: "-100vw"
      }}
      animate={{
        opacity: 1,
        x:0
      }}
      transition={{
        type: 'spring',
        stiffness: 100,
        // @ts-ignore
        ease: 'easeIn'
      }}
      exit={{
        opacity: 0,
        x: "-100vw"
      }}
      >

      
      <Stack spacing={8} mx={'auto'} maxW={'lg'} py={12} px={6}>
        <Stack align={'center'}>
          <Heading color={'white'} fontSize={'4xl'}>Sign in to your account</Heading>
        </Stack>
        <Box rounded={'lg'} bg={'white'} boxShadow={'lg'} p={8}>
          <Stack spacing={4}>
            <FormControl id="email">
              <FormLabel>Email address</FormLabel>
              <Input
                type="email"
                id="email"
                placeholder="name@email.com"
                name="email"
                value={state.email}
                onChange={handleChange}
              />
            </FormControl>
            <Stack spacing={10}>
              <FormControl id="password">
                <FormLabel>Password</FormLabel>
                <Input
                  type="password"
                  placeholder="Choose a password"
                  name="password"
                  value={state.password}
                  onChange={handleChange}
                />
              </FormControl>
              <Button
                bg={'custom.200'}
                color={'white'}
                _hover={{
                  bg: 'custom.300',
                }}
                isDisabled={validateForm()}
                onClick={handleSubmit}
              >
                Sign in
              </Button>
            </Stack>
              <RouterLink to='/register'>
                <Text align={'end'} color="blue.400">Create new account</Text>
              </RouterLink>
          </Stack>
        </Box>
      </Stack>
      </motion.div>
    </Flex>
  );
}