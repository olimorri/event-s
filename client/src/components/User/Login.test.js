import { render, screen, cleanup } from '@testing-library/react';
import Login from './Login';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';

const history = createMemoryHistory();

const LoginTest = () => (
  <Router history={history}>
    <Login />
  </Router>
);

afterEach(cleanup);

describe('Snapshot', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(<LoginTest />);

    expect(asFragment(<LoginTest />)).toMatchSnapshot();
  });
});

describe('Unit Testing', () => {
  test('renders Sign in component', () => {
    render(<LoginTest />);
    const registerTitle = screen.getByText(/Sign in to your account/i);
    expect(registerTitle).toBeInTheDocument();
  });

  test('renders email address input', () => {
    render(<LoginTest />);
    const emailInput = screen.getByLabelText(/Email address/i);
    expect(emailInput).toBeInTheDocument();
  });

  test('renders password input', () => {
    render(<LoginTest />);
    const emailInput = screen.getByLabelText(/Password/i);
    expect(emailInput).toBeInTheDocument();
  });

  test('renders sign in button', () => {
    render(<LoginTest />);
    const signInBtn = screen.getByRole('button', { name: /sign in/i });
    expect(signInBtn).toBeInTheDocument();
  });
});
