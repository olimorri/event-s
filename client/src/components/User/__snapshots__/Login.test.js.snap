// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-11cbxf6"
  >
    <div
      style="opacity: 0; transform: translateX(-100vw) translateZ(0);"
    >
      <div
        class="chakra-stack css-ffd652"
      >
        <div
          class="chakra-stack css-1xbaot1"
        >
          <h2
            class="chakra-heading css-tyi3eh"
          >
            Sign in to your account
          </h2>
        </div>
        <div
          class="css-3hziw7"
        >
          <div
            class="chakra-stack css-ppbuf4"
          >
            <div
              class="chakra-form-control css-1kxonj9"
              role="group"
            >
              <label
                class="chakra-form__label css-uvho5g"
                for="email"
                id="email-label"
              >
                Email address
              </label>
              <input
                class="chakra-input css-0"
                id="email"
                name="email"
                placeholder="name@email.com"
                type="email"
                value=""
              />
            </div>
            <div
              class="chakra-stack css-1re04b2"
            >
              <div
                class="chakra-form-control css-1kxonj9"
                role="group"
              >
                <label
                  class="chakra-form__label css-uvho5g"
                  for="password"
                  id="password-label"
                >
                  Password
                </label>
                <input
                  class="chakra-input css-0"
                  id="password"
                  name="password"
                  placeholder="Choose a password"
                  type="password"
                  value=""
                />
              </div>
              <button
                class="chakra-button css-1fvu2nh"
                disabled=""
                type="button"
              >
                Sign in
              </button>
            </div>
            <a
              href="/register"
            >
              <p
                class="chakra-text css-1o69x4v"
              >
                Create new account
              </p>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
